src/lib/stores/drips/index.ts

```diff
@@ -251,7 +263,8 @@ export default (() => {
   }
 
   async function topUp(workstream: EnrichedWorkstream, topUpDaiWei: bigint) {
-    const { provider } = get(internal);
+    const { provider } = get(internal) || {};
+    if (!provider) throw new Error('Connect the store first');
```

Comment:
Prolly a good idea to lint this:

```diff
diff --git a/.eslintrc.cjs b/.eslintrc.cjs
index 7e73783..7b9cc05 100644
--- a/.eslintrc.cjs
+++ b/.eslintrc.cjs
@@ -24,5 +24,6 @@ module.exports = {
   rules: {
     'no-console': 'error',
     '@typescript-eslint/no-unused-vars': 'error',
+    curly: ['error', 'multi-line', 'consistent']
   }
 };
```

https://eslint.org/docs/latest/rules/curly